client/src/main/java/org/spine3/base/MapStringifier.java

```diff
@@ -0,0 +1,188 @@
+
+    private static final char DEFAULT_ELEMENT_DELIMITER = ',';
+    private static final String ESCAPE_SEQUENCE = "\\";
+    private static final String KEY_VALUE_DELIMITER = ESCAPE_SEQUENCE + ':';
+
+    /**
+     * The delimiter for the passed elements in the {@code String} representation,
+     * {@code DEFAULT_ELEMENT_DELIMITER} by default.
+     */
+    private final String delimiter;
+    private final Class<K> keyClass;
+    private final Class<V> valueClass;
+
+    /**
+     * Creates a {@code MapStringifier}.
+     *
+     * <p>The {@code DEFAULT_ELEMENT_DELIMITER} is used for key-value
+     * separation in {@code String} representation of the {@code Map}.
+     *
+     * @param keyClass   the class of the key elements
+     * @param valueClass the class of the value elements
+     */
+    MapStringifier(Class<K> keyClass, Class<V> valueClass) {
+        super();
+        this.keyClass = keyClass;
+        this.valueClass = valueClass;
+        this.delimiter = ESCAPE_SEQUENCE + DEFAULT_ELEMENT_DELIMITER;
+    }
+
+    /**
+     * Creates a {@code MapStringifier}.
+     *
+     * <p>The specified delimiter is used for key-value separation
+     * in {@code String} representation of the {@code Map}.
+     *
+     * @param keyClass   the class of the key elements
+     * @param valueClass the class of the value elements
+     * @param delimiter  the delimiter for the passed elements via string
+     */
+    MapStringifier(Class<K> keyClass, Class<V> valueClass, String delimiter) {
+        super();
+        this.keyClass = keyClass;
+        this.valueClass = valueClass;
+        this.delimiter = ESCAPE_SEQUENCE + delimiter;
+    }
+
+    @Override
+    protected String toString(Map<K, V> obj) {
+        final String result = obj.toString();
+        return result;
+    }
+
+    @Override
+    protected Map<K, V> fromString(String s) {
+        final String[] buckets = s.split(Pattern.quote(delimiter));
+        final Map<K, V> resultMap = newHashMap();
+
+        for (String bucket : buckets) {
+            final Map.Entry<K, V> convertedBucket = convert(bucket);
+            resultMap.put(convertedBucket.getKey(), convertedBucket.getValue());
+        }
+        return resultMap;
+    }
+
+    private Map.Entry<K, V> convert(String bucketToConvert) {
+        final String[] keyValue = bucketToConvert.split(Pattern.quote(KEY_VALUE_DELIMITER));
+        checkKeyValue(keyValue);
+
+        final String key = keyValue[0];
+        final String value = keyValue[1];
+
+        try {
+            final K convertedKey = convert(keyClass, key);
+            final V convertedValue = convert(valueClass, value);
+            final Map.Entry<K, V> convertedBucket =
+                    new AbstractMap.SimpleEntry<>(convertedKey, convertedValue);
+            return convertedBucket;
+        } catch (Throwable e) {
+            throw newIllegalArgumentException("The exception is occurred during the conversion", e);
```

Comment:
It is tested with `throw_exception_when_passed_parameter_does_not_match_expected_format`